grouped-symbol-column/parseData.js

```diff
@@ -0,0 +1,130 @@
+            // range: getRanges(seriesNames, d),
+            groups: getGroups(seriesNames, d),
+            circleCats: getCircles(seriesNames, d, divisor),
+        }));
+
+        if (sort === 'descending') {
+            plotData.sort((a, b) =>
+                b.groups[sortOn].value - a.groups[sortOn].value);// Sorts biggest rects to the left
+        } else if (sort === 'ascending') {
+            plotData.sort((a, b) => a.groups[sortOn].value - b.groups[sortOn].value);
+        } // Sorts biggest rects to the left
+
+        return {
+            groupNames,
+            valueExtent,
+            seriesNames,
+            plotData,
+            data,
+        };
+    });
+}
+
+// a function that returns the columns headers from the top of the dataset, excluding specified
+function getSeriesNames(columns) {
+    const exclude = ['name', 'total'];
+    return columns.filter(d => (exclude.indexOf(d) === -1));
+}
+
+// a function that calculates the cumulative max/min values of the dataset
+function getMaxMin(values) {
+    const cumulativeMax = d3.sum(values.filter(d => (d > 0)));
+    const cumulativeMin = d3.sum(values.filter(d => (d < 0)));
+    return [cumulativeMin, cumulativeMax];
+}
+
+// a function to work out the extent of values in an array accross multiple properties...
+function extentMulti(data, columns) {
+    const ext = data.reduce((acc, row) => {
+        const values = columns.map(key => +row[key]);
+        const maxMin = getMaxMin(values);
+        const rowExtent = maxMin;
+        if (!acc.max) {
+            acc.max = rowExtent[1];
+            acc.min = rowExtent[0];
+        } else {
+            acc.max = Math.max(acc.max, rowExtent[1]);
+            acc.min = Math.min(acc.min, rowExtent[0]);
+        }
+        return acc;
+    }, {});
+    return [ext.min, ext.max];
+}
+
+function getRanges(seriesNames, el) { // eslint-disable-line
+    const ranges = [];
+    seriesNames.forEach((d, i) => {
+        ranges.push(+el[seriesNames[i]]);
+    });
+    return ranges;
+}
+
+function getCircles(seriesNames, el, divisor) {
+    const ranges = [];
+    let rangeTotal = 0;
+    seriesNames.forEach((d, i) => {
+        rangeTotal = ((+el[seriesNames[i]]) / divisor);
+        ranges.push(rangeTotal);
+    });
+
+    const numCircles = d3.range(el.total / divisor);
+
+    const circleCat = [];
+    let index = 0;
+    const stackIndex = [0];
+
+    seriesNames.forEach((obj, k) => {
+        if (k > 0) {
+            index += ranges[k - 1];
+            stackIndex.push(index);
+        }
+    });
+
+    for (let k = 0; k < seriesNames.length; k += 1) {
+        for (let i = 0; i < numCircles.length; i += 1) {
+            if (k < seriesNames.length - 1) {
+                if (i >= stackIndex[k] && i < stackIndex[k + 1]) {
+                    circleCat.push(seriesNames[k]);
+                }
+            } else if (i >= stackIndex[k]) {
+                circleCat.push(seriesNames[k]);
+            }
+        }
+    }
+
```

Comment:
I can't help but feel this might be more readable as some sort of combination of map and/or reduce, but maybe not.